tsl/test/expected/continuous_aggs_policy.out

```diff
@@ -219,13 +219,46 @@ INSERT INTO smallint_tab VALUES(5);
 INSERT INTO smallint_tab VALUES(10);
 INSERT INTO smallint_tab VALUES(20);
 CALL run_job(:job_id);
-SELECT * FROM mat_smallint;
+SELECT * FROM mat_smallint ORDER BY 1;
  a  | countb 
 ----+--------
   5 |      1
  10 |      1
 (2 rows)
 
+TRUNCATE table mat_smallint;
+--overwrite config file with out of range start/end interval to trigger overflow --
+-- the job should resolve the overflow and still run.
+SELECT hypertable_id as mat_id FROM _timescaledb_config.bgw_job where id = :job_id \gset
+\c :TEST_DBNAME :ROLE_SUPERUSER
+SELECT config FROM _timescaledb_config.bgw_job WHERE id = :job_id;
+                              config                               
+-------------------------------------------------------------------
+ {"end_interval": 0, "start_interval": 15, "mat_hypertable_id": 8}
+(1 row)
+
+UPDATE _timescaledb_config.bgw_job 
+SET config = jsonb_build_object('mat_hypertable_id', :mat_id,
```

Comment:
Is this a valid policy? If time is 20 (according to the now function), then the refresh interval will be `[20-90000, 20-900000] => [ -89980, -899980 ]`. If I am not mistaken, this is  an invalid range because the start interval is smaller than the end interval (i.e., the start of the refresh is after the end). The values are also, obviously, out-of-range for smallint. Which of these error cases are you testing?

I am also wondering what the expected behavior for such a bad policy would be. I guess it shouldn't even be possible to add such a policy, although here you are updating metadata directly so some error/validation at runtime would be expected, right? 

However, this doesn't seem to be the case when the policy is run below. Instead, it seems to update the aggregate. Can you explain what is going on?

Reply:
updated tests and modified  code based on our discussion